controllers/policy_controller.go

```diff
@@ -137,10 +138,26 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 				statusWriter, err := UpdateAWSObject(iamsvc, ins, DoNothingPreFunc)
 				statusWriter(ins, &policy, ctx, r.Status(), log)
 				if err != nil {
+					if aerr, ok := err.(awserr.Error); ok {
+						if aerr.Code() == awsiam.ErrCodeLimitExceededException {
+							// Remove oldests policy versions to make space for new one
+							err := CleanUpPolicyVersions(iamsvc, ins.ARN().String())
+							if err != nil {
+								log.Error(err, "error while cleaning up Policy versions during reconciliation")
+								return ctrl.Result{}, err
+							}
+						}
+					}
```

Comment:
This block of code will remove oldest policy if we are receiving "LimitExceeded" and will reconcile because spec.generation != status.observedGeneration